core/src/main/java/org/infinispan/reactive/publisher/impl/LocalPublisherManagerImpl.java

```diff
@@ -414,14 +419,18 @@ private int getNextSegment(PrimitiveIterator.OfInt segmentIter) {
          Function<? super Publisher<R>, ? extends CompletionStage<R>> finalizer) {
       AdvancedCache<K, V> cache = deliveryGuarantee == DeliveryGuarantee.AT_MOST_ONCE ? this.cache : remoteCache;
       return handleSpecificObjects(parallelPublisher, keysToInclude, keysToExclude, keyFlowable ->
-         keyFlowable.map(k -> {
-            CacheEntry<K, V> entry = cache.getCacheEntry(k);
-            if (entry == null) {
-               return NullCacheEntry.<K, V>getInstance();
-            }
-            return entry;
-         }).filter(e -> e != NullCacheEntry.getInstance())
-      , collator, finalizer);
+                  keyFlowable.concatMapMaybe(k -> {
+                     CompletableFuture<CacheEntry<K, V>> future = cache.getCacheEntryAsync(k);
+                     future = future.thenApply(entry -> {
+                        if (entry instanceof MVCCEntry) {
+                           // Scattered cache can return MVCCEntry instances
```

Comment:
What's the problem with `MVCCEntry` @wburns ?

Anyway I don't think it's specific to scattered cache, `getCacheEntry()` also returns a `MVCCEntry` in a transaction if a previous operation modified the key.

Reply:
If I remember correctly it was because the MVCC implementations are not supported currently to be marshalled.

The publisher doesn't use the context in the same way when looking up entries to find them in other cache modes.

Reply:
It would probably be more correct to use `InternalEntryFactory#createValue` instead, but this seemed sufficient.